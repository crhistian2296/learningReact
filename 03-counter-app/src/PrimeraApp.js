//import React, { Fragment } from "react";
import React from "react";
import PropTypes from "prop-types";

//solo se dara el valor por defecto si no se crea prop alguna al momento de llamar al componente
const PrimeraApp = ({ saludo, subtitulo }) => {
    /*     
    const saludo = {
        nombre: "Crhistian",
        edad: 25,
    };
    <pre>{JSON.stringify(saludo, null, 2)}</pre>
    */

    //console.log(props);
    //const saludo = "Hola mundo, este es mi primer componente";
    return (
        /*         
        <Fragment>
            <h1>Hola mundo, este es mi primer componente</h1>
            <p>Para mi primera aplicacion</p>
        </Fragment>
        */

        //Forma rapida
        <>
            <h1> {saludo} </h1>
            <p id="campanilla">{subtitulo}</p>
        </>
    );
};

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired,
};
PrimeraApp.defaultProps = {
    subtitulo: "Hola soy un subtitulo por default",
};

export default PrimeraApp;
