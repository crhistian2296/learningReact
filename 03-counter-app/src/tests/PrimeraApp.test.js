import React from "react";
import PrimeraApp from "../PrimeraApp";
import { shallow } from "enzyme";
// import { render } from "@testing-library/react";

describe("Pruebas en <PrimeraApp /> ", () => {
    /* 
    Solo funciona con import "@testing-library/jest-dom/extend-expect en estupTests.js
    test('Debe mostrar el Mensaje "Hola soy Goku"', () => {
        const saludo = "Hola soy Goku";

        const { getByText } = render(<PrimeraApp saludo={saludo} />);
        expect(getByText(saludo)).toBeInTheDocument();
    });
     */

    test("Deberia de mostrar <PirmeraApp /> correctamente", () => {
        const saludo = "Hola soy Goku";
        const wrapper = shallow(<PrimeraApp saludo={saludo} />);

        expect(wrapper).toMatchSnapshot();
    });
});
