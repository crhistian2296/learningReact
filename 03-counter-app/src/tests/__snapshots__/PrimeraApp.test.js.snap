// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp />  Deberia de mostrar <PirmeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
     
    Hola soy Goku
     
  </h1>
  <p>
     
    Hola soy un subtitulo por default
     
  </p>
</Fragment>
`;
